mne/viz/topomap.py

```diff
@@ -58,12 +60,17 @@ def _prepare_topo_plot(inst, ch_type, layout):
             raise ValueError("No channels of type %r" % ch_type)
 
         if layout is None:
-            from ..channels.layout import _find_topomap_coords
             pos = _find_topomap_coords(info, picks)
         else:
             names = [n.upper() for n in layout.names]
-            pos = [layout.pos[names.index(info['ch_names'][k].upper())]
-                   for k in picks]
+            try:
+                pos = [layout.pos[names.index(info['ch_names'][k].upper())]
+                       for k in picks]
+            except ValueError:  # special case, where pos info not in layout
+                logger.warning('Failed to locate %s channel positions from '
+                               'layout. Inferring channel positions from '
+                               'data.' % ch_type)
```

Comment:
Your suggestion won't work because ``k`` is inside a list comprehension and I can't think of a more elegant way of breaking the operation. Using for loop would be one option, but I like this one more because usually the operation should succeed anyways.
This here is done because sometimes the layout doesn't contain all the channel positions. For instance ``vectorview-all`` doesn't contain EEG channel positions in which case (only) the EEG positions are 'guessed' from the data.